src/ATen/test/broadcast_test.cpp

```diff
@@ -0,0 +1,171 @@
+#include "ATen/ATen.h"
+
+using namespace at;
+
+int main() {
+  Type & T = CPU(kFloat);
+
+  // 0) pre-req tests:
+  // can't expand empty tensor
+  try {
+    auto empty = T.randn({0});
+    empty.expand({3});
+    assert(false);
+  } catch(std::runtime_error &e) {}
+
+  // can't "expand" a Tensor to  Scalar
```

Comment:
nit: a Scalar to a Tensor?

Reply:
no, you can expand a scalar to a tensor.  You can't expand a tensor to a scalar.